framework/assets/yii.gridView.js

```diff
@@ -59,35 +85,38 @@
                 if (gridData[id] === undefined) {
                     gridData[id] = {};
                 }
+                if (gridEventHandlers[id] === undefined) {
+                    gridEventHandlers[id] = {};
+                }
 
                 gridData[id] = $.extend(gridData[id], {settings: settings});
 
+                var filterEvents = 'change.yiiGridView keydown.yiiGridView';
                 var enterPressed = false;
-                $(document).off('change.yiiGridView keydown.yiiGridView', settings.filterSelector)
-                    .on('change.yiiGridView keydown.yiiGridView', settings.filterSelector, function (event) {
-                        if (event.type === 'keydown') {
-                            if (event.keyCode !== 13) {
-                                return; // only react to enter key
-                            } else {
-                                enterPressed = true;
-                            }
+                initEventHandler($e, 'filter', filterEvents, settings.filterSelector, function (event) {
+                    if (event.type === 'keydown') {
+                        if (event.keyCode !== 13) {
+                            return; // only react to enter key
                         } else {
-                            // prevent processing for both keydown and change events
-                            if (enterPressed) {
-                                enterPressed = false;
-                                return;
-                            }
+                            enterPressed = true;
                         }
+                    } else {
+                        // prevent processing for both keydown and change events
+                        if (enterPressed) {
+                            enterPressed = false;
+                            return;
+                        }
+                    }
 
-                        methods.applyFilter.apply($e);
+                    methods.applyFilter.apply($e);
 
-                        return false;
-                    });
+                    return false;
+                });
             });
         },
 
         applyFilter: function () {
-            var $grid = $(this), event;
```

Comment:
Declaration of `event` variable is unnecessary here, it's used below in one specific place.